### PR TITLE
Update GooeyJr for Heroku

### PR DESCRIPTION
**Pull request by iojw**
This updates the repo so that it is ready for deployment on Heroku and includes the following changes:

1. Updates packages to the latest versions in package.json, including support for SASL authentication so that we can connect to freenode, along with added documentation in the README on how to do so.
2. Updates greeting script so that all new users entering the channel will get a welcome message. I previously limited it to only a few users for GCI testing.
3. Removes hubot-scripts.json as it is now deprecated
4. Updates external-scripts.json to remove any scripts that no longer work (translate, image, reminders, shipit) and also pugme for now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,3 +35,13 @@ To turn off the bot type `heroku ps:scale web=0`. You can also restart the bot b
 If you face any errors, make sure to check the logs by typing `heroku logs`. 
 
 Have fun!
+
+### Additional configuration for irc.freenode.net 
+
+If you are testing out GooeyJr, there is *no need* to set this up. Espernet will work fine for testing purposes.
+
+Unlike, Espernet, Freenode requires SASL authentication for any user connecting from  AWS (which Heroku uses). Therefore, we need to set the following environment variables to allow our bot to connect to Freenode from Heroku. You will need to register an account on Freenode using your email before setting this up.
```
Comment (mjuvekar7) [inline README.md]: No comma after "Unlike". Should be "Unlike Espernet, Freenode..."

**Comment by mjuvekar7** [comment]
The README needs a lot of work. It doesn't cover creating heroku apps and deploying GooeyJr on Heroku. Also there's a couple of grammatical mistakes here and there.

**Comment by iojw** [comment]
@mjuvekar7 Not sure if you've missed it, but the second section of the README covers creating and deploying GooeyJr on Heroku using the CLI. Also, would be good if you could point out the grammatical mistakes (apart from the extra comma) you are referring to individually by leaving comments at the appropriate line so I can fix them all at once 🙂.

Any other comments apart from the README?

**Comment by mjuvekar7** [comment]
Oops I was viewing the diff and not the actual README. Sorry, my bad.
I'm trying to get it running with heroku right now, so I'll try to leave comments on that in a bit.

**Comment by mjuvekar7** [comment]
There aren't any grammatical errors except for the extra comma. There were a couple of things that looked like errors in the diff, but in the context of the actual README it's perfect.
Sorry for that

File: README.md [hunk 2]
```diff
@@ -6,10 +6,8 @@ See the [Hubot Readme](https://github.com/github/hubot/blob/master/README.md) fo
 
 ## Environment setup  
 
-* Before beginning, ensure that you have both Ruby (https://www.ruby-lang.org/en/downloads/) and RubyGems (https://rubygems.org/pages/download) installed
-* Install `node.js` - https://docs.npmjs.com/getting-started/installing-node
+* Install heroku CLI using the instructions here - https://devcenter.heroku.com/articles/heroku-cli
 * Create an account on Heroku - https://www.heroku.com
-* Install the Heroku bindings: `gem install heroku`
 * Finally clone the project by entering `git clone https://github.com/MovingBlocks/GooeyJr` in your terminal
 * Alright, now that you have installed all of this, the fun part gets to happen
```
Comment (mjuvekar7) [inline README.md]: The fun part begins

"gets to happen" sounds a bit weird.

File: README.md (hunk 1 above)
Comment (mjuvekar7) [inline README.md]: Extra comma

File: README.md [hunk 3]
```diff
@@ -37,3 +35,13 @@ To turn off the bot type `heroku ps:scale web=0`. You can also restart the bot b
...
 Have fun!
+
+### Additional configuration for irc.freenode.net 
+
+If you are testing out GooeyJr, there is *no need* to set this up. Espernet will work fine for testing purposes.
+
+Unlike, Espernet, Freenode requires SASL authentication for any user connecting from  AWS (which Heroku uses). Therefore, we need to set the following environment variables to allow our bot to connect to Freenode from Heroku. You will need to register an account on Freenode using your email before setting this up.
+
+`heroku config:add HUBOT_IRC_USESASL="true"`  
+`heroku config:add HUBOT_IRC_USER="YOUR_USERNAME"`
```
Comment (iojw) [inline README.md]: This should be `HUBOT_IRC_USERNAME`

**Comment by mjuvekar7** [comment]
Works like a charm. Merge when you feel like it :)

**Comment by iojw** [comment]
Did some final tests and merged. Thanks for the review!

**Pull request merged**